ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java

```diff
@@ -729,8 +730,7 @@ public void onClose(
     ) {
       final String applicationId = query.getQueryApplicationId();
       Optional<String> topologyName = Optional.empty();
-      if (ksqlConfig.getBoolean(KsqlConfig.KSQL_SHARED_RUNTIME_ENABLED)
-          && !(query instanceof TransientQueryMetadata)) {
+      if (query instanceof BinPackedPersistentQueryMetadataImpl) {
```

Comment:
Since these changes are isomorphic, there's not really a good way to have a unit test for the change, right?

Reply:
Thats my thought. The mocks complained that the flag was an unnecessary mock, as we would expect.